src/verifast.ml

```diff
@@ -2835,11 +2835,24 @@ module VerifyProgram(VerifyProgramArgs: VERIFY_PROGRAM_ARGS) = struct
       in
       iter true init_list h
     in
+    let assume_field_ptr_limits struct_addr cont =
+      let rec iter fields =
+        match fields with
+        | [] -> cont ()
+        | (_, (_, _, _, Some offset, _)) :: fields ->
+          assume (mk_field_pointer_within_limits struct_addr offset) @@ fun () -> iter fields
+        | _ :: fields -> iter fields
+      in
+      iter fields
+    in
     let init_fields delegated h struct_addr this_type init_list env ghostenv leminfo sizemap current_thread cont =
       (* when a delegated ctor was called, no initializeation is needed for the fields *)
       if delegated then cont h else
+      begin fun cont ->
+        assume_field_ptr_limits struct_addr cont
+      end @@ fun () ->
       let init_field h field_name field_type value struct_addr cont =
-        assume_field h struct_name field_name field_type Real struct_addr value real_unit cont
+        assume_field h struct_name field_name field_type Real struct_addr value real_unit false cont

```

Comment:
The extra boolean argument is to avoid re-assuming the same fact again? The marginal performance benefit seems less than the code complexity cost...